hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -253,6 +253,14 @@ public void persistToken(Token token) {
 			}
 			accountStore.persistAccount(token.getTreasury());
 		}
+		if (token.hasBurnedUniqueTokens()){
+			for (final UniqueToken uniqueToken : token.burnedUniqueTokens()) {
+				final var uniqueId = new MerkleUniqueTokenId(new EntityId(uniqueToken.getTokenId()), uniqueToken.getSerialNumber());
+				uniqueTokens.get().remove(uniqueId);
+			}
+			accountStore.persistAccount(token.getTreasury());
```

Comment:
Michael had a comment about persisting the account from the TypedTokenStore. 

https://github.com/hashgraph/hedera-services/pull/1615#discussion_r653181862

Lets resolve his comment and change how we do the mint as-well

Reply:
So the accountStore.persist should be called in the transitionLogic?